server.js

```diff
@@ -0,0 +1,186 @@
+const express = require('express');
+const cors = require('cors');
+const bodyParser = require('body-parser');
+const mongoose = require('mongoose');
+
+const User = require('./user');
+const BlogPost = require('./blogPost');
+
+
+const server = express();
+server.use(bodyParser.json());
+server.use(cors());
+
+// Users
+server.post('/users', (req, res) => {
+    const { fullName, email, age, password } = req.body;
+    if(!fullName || !email || !password ) {
+        res.status(422).send({error: "You are missing one of the important components to create a user"})
+        return;
+    }
+    const newUser = new User({ fullName, email, age, password });
+    newUser.save((err, user) => {
+        if (err) {
+            res.status(500).send(err);
+            return;
+        } else {
+            res.json(user);
+        }
+    })
+});
+
+server.get('/users', (req, res) => {
+    User
+        .find()
+        .exec((err, users) => {
+            if (err) {
+                res.status(500).send(err);
+                return;
+            } else {
+                res.json(users);
+            }
+
+        })
+});
+
+server.get('/users/:id', (req, res) => {
+    User
+        .findById(req.params.id)
+        .exec((err, user) => {
+            if (err) {
+                res.status(500).send(err);
+            } else {
+                res.json(user);
+            }
+        })
+});
+
+server.put('/users/:id', (req, res) => {
+    User
+        .findByIdAndUpdate(req.params.id, 
+            { $set: req.body }, 
```

Comment:
this is actually pretty bad practices. You should pull off your req.body before you just do any data saving. This can easily lead to security issues :) Please refactor to pull of the intended edits from the req.body before $setting it in the document.

Reply:
Thank you! I didn't know this was bad practice. I'll definitely look into the security risks here.